File: data/multilang_comparison/program_cpp_vector/src/func40.c
#include "program_cpp_vector.h" 
array_t* func40(array_t_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array17;
   if (pCounter > 0) {
      array17 = vars->data[--pCounter];
      array17->refC++;
      DEBUG_COPY(array17->id);
   } else {
      array17 = (array_t*)malloc(sizeof(array_t));
      array17->size = 84;
      array17->refC = 1;
      array17->id = 17;
      array17->data = (unsigned int*)malloc(array17->size*sizeof(unsigned int));
      memset(array17->data, 0, array17->size*sizeof(unsigned int));
      DEBUG_NEW(array17->id);
   }
   unsigned int loop7 = 0;
   unsigned int loopLimit7 = (rand()%loopsFactor)/5 + 1;
   for(; loop7 < loopLimit7; loop7++) {
      for (int i = 0; i < array17->size; i++) {
         if (array17->data[i] == 36) { 
            return array17;
         }
      }
   }
   for (int i = 0; i < array17->size; i++) {
      array17->data[i]--;
   }
   for (int i = 0; i < array17->size; i++) {
      if (array17->data[i] == 29) { 
         return array17;
      }
   }
   return array17;
}

File: data/multilang_comparison/program_go/program_go/src/func40.go
package main
import "math/rand"
var _ = rand.Intn
func Func40(vars *ArrayParam, loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array17 *Array
   if pCounter > 0 {
       array17 = vars.Data[pCounter-1]
       pCounter--
       array17.RefC++
   } else {
       array17 = &Array{
           Size: 84,
           RefC: 1,
           Id: 17,
           Data: make([]uint32, 84),
       }
   }
   var loop7 int = 0;
   var loopLimit7 int = (50)/5 + 1;
   for ; loop7 < loopLimit7; loop7++ {
      for i := 0; i < array17.Size; i++ {
          if array17.Data[i] == 36 {
              return array17
          }
      }
   }
   for i := 0; i < array17.Size; i++ {
       array17.Data[i]--
   }
   for i := 0; i < array17.Size; i++ {
       if array17.Data[i] == 29 {
           return array17
       }
   }
   return array17;
}

